File: Docs/UT7-DesarrolloDeAplicacionesAndroid/Example01/app/src/main/java/com/raks/adding/MainActivity.java
package com.raks.adding;

import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button button = findViewById(R.id.add_button);
        button.setOnClickListener(v -> {
            EditText firstNumberEditText  = findViewById(R.id.first_number);
            String   firstNumberText      = firstNumberEditText.getText().toString();
            int      firstNumber          = Integer.parseInt(firstNumberText);
            EditText secondNumberEditText = findViewById(R.id.second_number);
            String   secondNumberText     = secondNumberEditText.getText().toString();
            int      secondNumber         = Integer.parseInt(secondNumberText);
            ((TextView) findViewById(R.id.result)).setText(String.valueOf(firstNumber + secondNumber));
        });
    }
}

File: Tests/P9-ConstraintLayout/app/src/main/java/com/danielmunoz/android/MainActivity.kt
package com.danielmunoz.android

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    private lateinit var  first: EditText
    private lateinit var second: EditText
    private lateinit var result: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        first  = findViewById(R.id.first_number)
        second = findViewById(R.id.second_number)
        result = findViewById(R.id.result)

        findViewById<Button>(R.id.add).setOnClickListener { add() }
        findViewById<Button>(R.id.clear).setOnClickListener { clear() }
    }

    private fun add() {
        val firstNumber  = first.text.toString().toIntOrNull()  ?: 0
        val secondNumber = second.text.toString().toIntOrNull() ?: 0
        result.text = getString(R.string.result, firstNumber + secondNumber)
    }

    private fun clear() {
        first.text.clear()
        second.text.clear()
        result.text = getString(R.string.default_result)
    }
}